Move some filesystems to new api

samikama opened the pull request:
This PR is part of introducing the C++ api layer changes for Transactional FileSystem support. It requires PR #41433 to be merged before being merged. There will be follow up PRs to add other filesystems and re-enable override keywords once base class FileSystem method signature is changed.

samikama committed:
Move some filesystems to new api and workaround compiler flags.

mihaimaruseac commented on tensorflow/c/experimental/filesystem/modular_filesystem.h:
```diff
@@ -61,34 +61,34 @@ class ModularFileSystem final : public FileSystem {
   Status NewRandomAccessFile(
       const std::string& fname,
-      std::unique_ptr<RandomAccessFile>* result) override;
+      std::unique_ptr<RandomAccessFile>* result, TransactionToken* token=nullptr) /*override*/;
```
Why the removal of `override`?

mihaimaruseac replied:
Oh, because base class is not changed yet and we cannot change it before these change. Makes sense

gbaned commented:
@samikama Can you please check build failures. Thanks!

samikama commented:
@mihaimaruseac I modified these changes to workaround compiler flags. I reverted override comments as well since compiler flag is enforcing override anyway. @gbaned  this PR will be able to compile after #41433  is merged. I will open new PRs for other FileSystem derivatives as well.

samikama commented:
Since we commented extra parameters, this PR doesn't depend on #41433  anymore and can be merged at any time.
